base/util/merge.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.)
+
+package util
+
+import (
+	"fmt"
+
+	"github.com/coreos/fcct/translate"
+
+	"github.com/coreos/ignition/v2/config/merge"
+)
+
+// MergeTranslatedConfigs merges a parent and child config and returns the
+// result.  It also generates and returns the merged TranslationSet by
+// mapping the parent/child TranslationSets through the merge transcript.
+func MergeTranslatedConfigs(parent interface{}, parentTranslations translate.TranslationSet, child interface{}, childTranslations translate.TranslationSet) (interface{}, translate.TranslationSet) {
+	// mappings:
+	//   left:  parent or child translate.TranslationSet
+	//   right: merge.Transcript
+
+	// merge configs
+	result, right := merge.MergeStructTranscribe(parent, child)
+
+	// merge left and right mappings into new TranslationSet
+	if parentTranslations.FromTag != childTranslations.FromTag || parentTranslations.ToTag != childTranslations.ToTag {
```

Comment:
Does this prevent us from being able to potentially add merged FCCT configs in the future?

Reply:
It shouldn't.  If we merge FCCs after translating them, both parent and child will have TranslationSets mapping `yaml` -> `json`, as they do now.  If we merge FCCs before translating them, this function won't be useful anyway, since it assumes there's already been one level of translation before the merge occurs.